### PR TITLE
Implement `describe.skip`

### PR DESCRIPTION
**Pull request by blackmann**

### CODE REVIEW

File: src/bun.js/builtins/cpp/BundlerPluginBuiltins.cpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ namespace WebCore {
 const JSC::ConstructAbility s_bundlerPluginRunOnResolvePluginsCodeConstructAbility = JSC::ConstructAbility::CannotConstruct;
 const JSC::ConstructorKind s_bundlerPluginRunOnResolvePluginsCodeConstructorKind = JSC::ConstructorKind::None;
 const JSC::ImplementationVisibility s_bundlerPluginRunOnResolvePluginsCodeImplementationVisibility = JSC::ImplementationVisibility::Public;
-const int s_bundlerPluginRunOnResolvePluginsCodeLength = 3593;
+const int s_bundlerPluginRunOnResolvePluginsCodeLength = 3619;
```
Comment (blackmann) [inline src/bun.js/builtins/cpp/BundlerPluginBuiltins.cpp]: This and other changes in this file were auto-generated. I suppose that's expected?

File: test/js/bun/test/test-test.test.ts [hunk 2]
```diff
@@ -2612,3 +2612,51 @@ it("should return non-zero exit code for invalid syntax", async () => {
...
+  });
+
+  describe.skip("should skip", () => {
+    it("should throw", () => {
+      throw new Error("This should not throw. `.skip` is broken");
+    });
+  });
+});
+
+describe.skip("skip test outer", () => {
```
Comment (Jarred-Sumner) [inline test/js/bun/test/test-test.test.ts:2634]: can you add a test that checks we also skip `beforeAll`, `afterAll`, `beforeEach`, and `afterEach` (assuming Jest does that too)?

File: src/bun.js/builtins/cpp/BundlerPluginBuiltins.cpp (hunk 1 above)
Comment (Jarred-Sumner) [reply src/bun.js/builtins/cpp/BundlerPluginBuiltins.cpp]: Yeah this was fixed on main before you rebased last

File: src/bun.js/test/jest.zig [hunk 3]
```diff
@@ -3635,7 +3635,7 @@ pub const DescribeScope = struct {
         this.pending_tests.unset(test_id);
 
         if (!skipped) {
-            const afterEach = this.execCallback(globalThis, .afterEach);
+            const afterEach = this.runCallback(globalThis, .afterEach);
```
Comment (blackmann) [inline src/bun.js/test/jest.zig:3638]: Incidentally resolves: #2528

Comment (Jarred-Sumner) [reply src/bun.js/test/jest.zig:3638]: nice

**Comment by Jarred-Sumner** [comment]
This looks great. Thank you. There will be some unrelated test failures, don't worry about those. Let's merge after tests run

**Review by Jarred-Sumner — APPROVED** [review]

**Pull request merged**